packages/webcrypto-ed25519-polyfill/src/secrets.ts

```diff
@@ -137,17 +137,24 @@ function getPublicKeyBytes(key: CryptoKey): Uint8Array {
 export function exportKeyPolyfill(format: 'jwk', key: CryptoKey): JsonWebKey;
 export function exportKeyPolyfill(format: KeyFormat, key: CryptoKey): ArrayBuffer;
 export function exportKeyPolyfill(format: KeyFormat, key: CryptoKey): ArrayBuffer | JsonWebKey {
+    if (key.extractable === false) {
+        throw new DOMException('key is not extractable', 'InvalidAccessException');
+    }
     switch (format) {
         case 'raw': {
-            if (key.extractable === false) {
-                throw new DOMException('key is not extractable', 'InvalidAccessException');
-            }
             if (key.type !== 'public') {
                 throw new DOMException(`Unable to export a raw Ed25519 ${key.type} key`, 'InvalidAccessError');
             }
             const publicKeyBytes = getPublicKeyBytes(key);
             return publicKeyBytes;
         }
+        case 'pkcs8': {
+            if (key.type !== 'private') {
+                throw new DOMException(`Unable to export a pkcs8 Ed25519 ${key.type} key`, 'InvalidAccessError');
+            }
```

Comment:
So it doesn't let you export public keys with pkcs8?

Reply:
Nope, you have to use `raw` for that. Node 18:
```
> const keypair = await crypto.subtle.generateKey('Ed25519', true, ['sign', 'verify'])
> await crypto.subtle.exportKey('pkcs8', keypair.publicKey)
Uncaught:
DOMException [InvalidAccessError]: Unable to export a pkcs8 Ed25519 public key
    at new DOMException (node:internal/per_context/domexception:53:5)
    at __node_internal_ (node:internal/util:520:10)
    at exportKeyPkcs8 (node:internal/crypto/webcrypto:339:9)
    at SubtleCrypto.exportKey (node:internal/crypto/webcrypto:454:26)
    at REPL467:1:53
```

Reply:
Strange API innit?

Reply:
Sure is! Delightful to polyfill :p